app/src/main/java/woowacourse/payments/ui/newcard/component/SelectedBankBottomSheet.kt

```diff
@@ -0,0 +1,52 @@
+package woowacourse.payments.ui.newcard.component
+
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.tooling.preview.Preview
+import woowacourse.payments.ui.newcard.state.CardStateHolder
+import woowacourse.payments.ui.newcard.uiModel.BankTypeUiModel
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SelectedBankBottomSheet(
+    state: CardStateHolder,
+    onDismissRequest: () -> Unit,
+) {
+    var selectedBankCard by rememberSaveable { mutableStateOf(BankTypeUiModel.NOT_SELECTED) }
+
+    val bottomSheetState = rememberModalBottomSheetState(
+        skipPartiallyExpanded = false,
+        confirmValueChange = { false },
+    )
+
+    LaunchedEffect(key1 = selectedBankCard) {
+        if (selectedBankCard != BankTypeUiModel.NOT_SELECTED) {
+            bottomSheetState.hide()
+            onDismissRequest()
+        }
+    }
+    ModalBottomSheet(
+        sheetState = bottomSheetState,
+        onDismissRequest = { onDismissRequest() },
+    ) {
+        SelectedBankRow(
+            selectedBank = { selectedBank ->
+                state.changeBankType(selectedBank)
+                selectedBankCard = selectedBank
```

Comment:
```suggestion
                onDismissRequest()
```

와 같이 바로 `DismissRequest()`를 호출해도 바텀 시트는 닫히는 것 같아요!
 `selectedBankCard = selectedBank`를 통해 `LaunchedEffect(key1 = selectedBankCard)`를 사용하신 이유가 궁금합니다! 약간의 차이점도 있는 것 같더라구요

Reply:
1. 첫번째 이유
LMS에 나와있어서 그대로 사용하고 있었습니당

2. 두번째 이유
저의 의도는 `state`가 달라질 때마다, 모달이 닫히고 열리는 것을 의도했습니다. `onDismissRequest()`는 그냥 모달을 닫아주는 역할만 하고, `state`와 동적으로 움직이지 않는다고 생각했습니다. `stateHolder`를 쓴다면, `state`를 활용하자는 의미에서도 `launchedEffect`를 사용했습니다.

메다를 어떻게 구현하시고 둘의 차이는 뭐라고 생각하세요?

Reply:
맞아용 닫히는 이벤트가 상위로 전달되고 state가 바뀌어서 닫히는 방식이 저도 좋은 것 같아요

둘의 차이점에 관한 질문은 제가 잘못 남깃 것 같아서... [다시 남길게용](https://github.com/woowacourse/android-payments/pull/67/files#r2363463540)